src/ws/cockpitauth.c

```diff
@@ -1156,6 +1165,47 @@ cockpit_auth_check_cookie (CockpitAuth *self,
     }
 }
 
+/*
+ * returns TRUE if auth can proceed, FALSE otherwise.
+ * dropping starts at connection max_startups_begin with a probability
+ * of (max_startups_rate/100). the probability increases linearly until
+ * all connections are dropped for startups > max_startups
+ */
+
+static gboolean
+cockpit_auth_can_start_auth (CockpitAuth *self)
```

Comment:
In addition, can the caller of this function be responsible for incrementing self->startups? That way there'll be symmetry in where it's incremented and decremented. Otherwise this causes a lot of questions when reading teh code.